### PR TITLE
Smokening

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
1) All smoke can now flow through any type of smoke unimpeded.

This was previously limited to just defiler smoke. Gone will be the days of one stray smoke tile blocking an entire boiler glob through a doorway.

2) Smoke that crosses paths with smoke of the same type will merge with the existing smoke instead of stacking more smoke on top of it

~~This behaves like fire. It will stack in duration up to a maximum of x2 the original duration~~, which means you don't get cucked by trying to resmoke 0.5 seconds too early and having the old smoke completely block it.
Edit: Now just goes up to the existing duration or duration of the incoming smoke, so its a little less oppressive to have 5 acid globs thrown on top of you.

This is a fairly significant balance change that positively and negatively effects both sides, so I'd want to TM this to see how it actually plays out, although I suspect it will be Xenos that gain the most from this.
## Why It's Good For The Game
Less fucky smoke interactions, behaves more like you would expect.
## Changelog
:cl:
balance: Smoke can now flow past other existing smoke clouds
/:cl:

### CODE REVIEW

**Comment by DeltaFire15** [comment]
I don't know what you mean with advantages for both sides, there isn't really any offensive smoke that marines use where stacking would be useful. On the other hand, for xenos, this completely unchains all smoke castes and turns something that could be called powerful, and sometimes pretty daunting, into definitely oppressive, aswell as removes the only counterplay marines had for heavy use of smoke deploying castes.

**Comment by Lumipharon** [comment]
> I don't know what you mean with advantages for both sides, there isn't really any offensive smoke that marines use where stacking would be useful. On the other hand, for xenos, this completely unchains all smoke castes and turns something that could be called powerful, and sometimes pretty daunting, into definitely oppressive, aswell as removing the only counterplay marines had for heavy use of smoke deploying castes.

You can now chain smokes together, to extend duration of plasma smoke, or plasma smoke while in other smoke etc. Previously you would effectively waste your smoke if it was blocked by any other smoke.

**Comment by Losenis** [comment]
>Quad defiler stacking all the chems in one go

image

**Comment by Lumipharon** [comment]
Defilers already can do that.

**Comment by Pariah919** [comment]
honestly this is pretty cringe and kills a ton of emergent gameplay, I dislike this

**Comment by Araulius** [comment]
welp, standard smoke nades are kill

**Comment by Autisem** [comment]
This does change quiet alot of balance around not just boiler but stacking as in general.

Would still rather have this be an toggolable option but its would be interseting

**Comment by novaepee** [comment]
If this gets merge, just make a marine tool that allow to block xenomorph gas. Simple.

**Comment by CCorvet** [comment]
> This just makes more sense and should've been default behaviour all along, esp considering smoke is literally a requirement for splitting up ungaballs; though, it could be cool to have some kind of specialised smoke dispersal tool/machinery on the marine side as a counter beyond "take mimir lol"

big fan, boom

**Comment by Ragecrabbo** [comment]
### **_The fog is coming_**

**Comment by Losenis** [comment]
>Defilers already can do that.

Can they? I thought if 4 defilers vented gas in the same area they didn't overlap.

**Comment by shelteredfreak** [comment]
All xeno gasses can expand inside eachother, acid, neuro, and defiler gasses. This PR would make stacking gasses more effective due to increasing the duration, as well as allowing Xeno gasses to pass through marine gasses like cloak, smoke, and tangle. 

P much a straight buff to Defiler and Boiler. Which is a good or bad thing depending on how you view balance. I'm biased and would like to see this go through, or at the very least be TM'd.

**Comment by Lumipharon** [comment]
> All xeno gasses can expand inside eachother, acid, neuro, and defiler gasses. This PR would make stacking gasses more effective due to increasing the duration, as well as allowing Xeno gasses to pass through marine gasses like cloak, smoke, and tangle.
> 
> P much a straight buff to Defiler and Boiler. Which is a good or bad thing depending on how you view balance. I'm biased and would like to see this go through, or at the very least be TM'd.

Specifically, defiler gases, and defiler gases only, allow OTHER gases to flow through them. So defilers can stack gas, and a boiler can shoot into defiler gas etc, but a defiler can gas into a boiler cloud.

**Review by QualityVan — APPROVED** [review]
Agree that this kills the only use for smoke grenades in current hvx, but it's cleaner as a system because the way smoke determined passthrough was weird and this is just more intuitive.

**Comment by QualityVan** [comment]
It's a little out of scope for this particular pr, but what if generic smoke grenade smoke got a flag to destroy any other clouds in its process?

**Comment by Lumipharon** [comment]
> It's a little out of scope for this particular pr, but what if generic smoke grenade smoke got a flag to destroy any other clouds in its process?

Easy enough, but yeah I'd rather do any other changes in a future pr.

Also normal smoke isn't even useful for that now, since you can just use plasma or cloak smoke for the same effect.

**Comment by JackTheJackhammer** [comment]
gonna be boiler hell

**Comment by Strayed-Raven** [comment]
This gives absolutely no counter to mass spam of Xeno gas. Boiler and Defiller stacking is just ggnore now because of this PR. I would say give some kind of weight to each gas so it does not fully bleed into every other smoke so marines smoke can at least still be a shield. otherwise Mimir should never be nerfed after this.

**Comment by SiegeB0t** [comment]
I am a fan of the smoke denial interactions and this reduces boiler counter play options to just tanking boiler globs with your face. Perhaps we can fix this by triggering boiler globs prematurely on explosion or bullet hit.

**Comment by JackTheJackhammer** [comment]
Just played a round with this and the gas spam is iNSANE. Its 1914 out there!

File: code/game/objects/effects/effect_system/smoke.dm [hunk 1]
```diff
@@ -170,7 +170,7 @@
 	amount = parent.amount-1
 	origin = parent.origin
 	if(merge)
-		lifetime = min((parent.lifetime + lifetime), initial(lifetime) * 2)
+		lifetime = min(parent.lifetime, lifetime)
```
Comment (QualityVan) [inline code/game/objects/effects/effect_system/smoke.dm:173]: Shouldn't this be max?

Comment (Lumipharon) [reply code/game/objects/effects/effect_system/smoke.dm:173]: Yes, honk

**Comment by Derrick2Electricboogaloo** [comment]
Damn smoke nades are useless now.

**Pull request merged**